src/standalone-js/src/deckgl.js

```diff
@@ -0,0 +1,178 @@
+/* global window, document */
+/* eslint-disable max-statements */
+import Mapbox from './mapbox';
+
+const CANVAS_STYLE = {
+  position: 'absolute',
+  left: 0,
+  top: 0,
+  width: '100%',
+  height: '100%'
+};
+
+/**
+ * @params container (Element) - DOM element to add deck.gl canvas to
+ * @params map (Object) - map API. Set to falsy to disable
+ * @params controller (Object) - Controller class. Leave empty for auto detection
+ */
+class DeckGL {
+  constructor(props = {}) {
+    if (typeof document === 'undefined') {
+      // Not browser
+      return;
+    }
+
+    const {container = document.body} = props;
+    const width = container.clientWidth;
+    const height = container.clientHeight;
+
+    // Add DOM elements
+    const containerStyle = window.getComputedStyle(container);
+    if (containerStyle.position === 'static') {
+      container.style.position = 'relative';
+    }
+
+    const mapCanvas = document.createElement('div');
+    container.appendChild(mapCanvas);
+    Object.assign(mapCanvas.style, CANVAS_STYLE);
+
+    const deckCanvas = document.createElement('canvas');
+    container.appendChild(deckCanvas);
+    Object.assign(deckCanvas.style, CANVAS_STYLE);
+
+    const {map, controller} = props;
+
+    this._getViewState(props);
+    const isMap = Number.isFinite(props.viewState.latitude);
+    const isOrbit = Number.isFinite(props.viewState.distance);
+
+    // Update viewport dimensions
+    Object.assign(props, {
+      width,
+      height
+    });
+
+    this._deck = new DeckGL.experimental.DeckGLJS(
```

Comment:
`DeckGLJS` is deprecated in favor of `DeckGL` in 5.2. It may even be an official export.